R/text_tools.R

```diff
@@ -119,58 +81,269 @@ extract_location <- function(v) {
+#' }
 #' @export
-extract_split <- function(text, splitsign = "\\.") {
-  t <-  strsplit(as.character(text), splitsign)
+split_text <- function(v, splitsign = "\\.") {
+  doc_id <- sentence <- token <- text <- NULL
+  if (inherits(v, "data.frame")) {
+    if (!"doc_id" %in% names(v)) {
+      stop("Please declare a text vector or an annotated data frame.")
+    }
+    if ("sentence" %in% names(v)) {
+      v <- dplyr::group_by(doc_id) %>%
+        dplyr::summarise(text = paste(sentence, collapse = " ")) %>%
+        dplyr::select(text)
+    } else if ("token_id" %in% names(v)) {
+      v <- dplyr::group_by(doc_id) %>%
+        dplyr::summarise(text = paste(token, collapse = " ")) %>%
+        dplyr::select(text)
+    }
+  }
+  t <-  strsplit(as.character(v), splitsign)
   # Add attribute for the number of divisions
   for(i in seq_len(length(t))) {
     attr(t[[i]], "Parts") <- paste0("Number of parts = ", lengths(t[i]))
   }
   t
 }
 
-#' Extract text matches
+#' Load text from PDFs
 #'
-#' Get texts in which only certain "matches" occur.
-#' @param text A text variable
-#' @param match A regex match for a word(s) or expression.
-#' For multiple words, please use "|" to divide them.
-#' @param invert Do you want texts without certain matches to be returned?
-#' By default FALSE.
-#' @param ignore.case Should case be ignored?
-#' By default, TRUE.
-#' @importFrom purrr map_chr
-#' @return A list of matches of the same length as text variable
-#' @examples
-#' text <- c("This function was created on the 29 September 2021",
-#' "Today is October 12, 2021")
-#' extract_match(text, "October")
+#' @param path The path to a PDF file or a folder containing multiple PDFs.
+#' @return A list of texts.
+#' @importFrom pdftools pdf_ocr_text
 #' @export
-extract_match <- function(text, match, invert = FALSE, ignore.case = TRUE) {
-  if (invert == TRUE & ignore.case == FALSE) {
-    t <- lapply(text, function(x) grep(match, x, value = TRUE, ignore.case = FALSE, invert = TRUE))
-  } else if (invert == TRUE & ignore.case == TRUE) {
-    t <- lapply(text, function(x) grep(match, x, value = TRUE, ignore.case = TRUE, invert = TRUE))
-  } else if (invert == FALSE & ignore.case == FALSE) {
-    t <- lapply(text, function(x) grep(match, x, value = TRUE, ignore.case = FALSE))
+load_pdf <- function(path) {
+  thisRequires("tesseract")
+  if (grepl(".pdf", path)) {
+    out <- paste(pdftools::pdf_ocr_text(x), collapse = " ")
   } else {
-    t <- lapply(text, function(x) grep(match, x, value = TRUE, ignore.case = TRUE))
+    out <- list()
+    for (x in list.files(path = path, full.names = TRUE,
+                         pattern = "*.pdf")) {
+      out[[basename(x)]] <- paste(pdftools::pdf_ocr_text(x), collapse = " ")
+    }
+  }
+  out
+}
+
+#' Annotate text with NLP
+#'
+#' This function builds upon `spacyr::spacy_parse` function to annotate texts.
+#' @param v Text vector
+#' @param level Do you want to parse words or sentences? Words by default.
```

Comment:
Like above, formulation suggests a logical argument. At least rewrite as "words" and "sentences" or "sentence"